### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by stewartTS**
https://trello.com/c/loDrOJWI/8-navbar-component

### CODE REVIEW

File: WhaleSpotting/react-app/src/components/Layout.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
 export default function Layout({children}: {children: JSX.Element | JSX.Element[]}): JSX.Element {
 
     return (
         <div>
-            <NavMenu />
+            <Navbar />
             <Container>
                 {children}
             </Container>
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/Layout.tsx]: I think we can delete this file

File: WhaleSpotting/react-app/src/components/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export default function Navbar(): JSX.Element {
+    const location = useLocation();
+    const removeSlash = location.pathname.slice(1);
+    const pageName = removeSlash === "" ? "Home" : removeSlash.substr(0, 1).toUpperCase() + removeSlash.substr(1).toLowerCase();
+    const [currentPage, setCurrentPage] = useState(pageName);
+
+    function HandleLinkClick(currentPage: string) {
+
+        setCurrentPage(currentPage);
+        console.log(currentPage);
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/Navbar.tsx]: Can you remove the console log and add a lint rule to not allow any console logs

File: WhaleSpotting/react-app/src/components/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    const loggedIn = false;
+
+    useEffect(() => {
+        setCurrentPage(pageName);
+    }, [location]);
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/Navbar.tsx:18]: new line after this please

File: WhaleSpotting/react-app/src/components/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return (
+        <div className="header">
+            <nav className="nav-bar">
+                <div className="group1">
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/Navbar.tsx]: I would suggest renaming group1 and group2 to something more descriptive

File: WhaleSpotting/react-app/src/components/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                    <Link to="./Home" data-testid="home"
+                        className={currentPage === "Home" ? "nav-link-selected" : "navbar-link"}
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/Navbar.tsx]:
```suggestion
                        className={getClassName(currentPage, "Home")}
```
It may be nice to change all of these classnames to this so that you can have a bit of shortened code :)

File: WhaleSpotting/react-app/src/index.scss [hunk 6]
```diff
@@ -1,13 +1,11 @@
...
+  font-family: $fontFamily;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
 }
 
-code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
-    monospace;
-}
+
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/index.scss]: extra new line

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 7]
```diff
@@ -0,0 +1,68 @@
+@import './constants.scss';
+
+* {
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: What's this meant to be for? We don't want to be adding global styling

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    margin: 0;
+    padding: 0;
+}
+
+.header {
+    position: sticky;
+    top: 0;
+    width: 100%;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss:6]: Not needed I think

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    background-color: $primaryColour;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: $headerHeight;
+    z-index: 10;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: Why was this added?

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.nav-bar {
+    display: flex;
+    width: 100%;
+    max-width: 1024px;
+    flex-direction: row;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss:18]: Not needed

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 11]
```diff
@@ -0,0 +1,68 @@
+@import './constants.scss';
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: this needs to be capitalised
```suggestion
@import './Constants.scss';
```

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    margin-left: 5%;
+    margin-right: 5%;
+    padding: 3%;
+    white-space: nowrap;
+    color: white;
+    text-decoration: none;  
+    font-size: calc(10px + 1vmin);  
+}
+
+.navbar-link:hover {
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]:
```suggestion
    &:hover {
```
You can merge these slightly

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    flex-direction: row;
+    justify-content: space-between;
+}
+
+.navbar-link {
+    margin-left: 5%;
+    margin-right: 5%;
+    padding: 3%;
+    white-space: nowrap;
+    color: white;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: This should be using the constant defined in constants :)

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+    display: flex;
+    width: 100%;
+    max-width: 1024px;
+    flex-direction: row;
+    justify-content: space-between;
+}
+
+.navbar-link {
+    margin-left: 5%;
+    margin-right: 5%;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: You can merge these
```suggestion
    margin: 0 5%;
```

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+    padding: 3%;
+    white-space: nowrap;
+    color: white;
+    text-decoration: none;  
+    font-size: calc(10px + 1vmin);
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: I'd suggest using a constant for this

File: WhaleSpotting/react-app/src/components/Navbar.tsx [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+    const loggedIn = false;
+
+    useEffect(() => {
+        setCurrentPage(pageName);
+    }, [location]);
+    return (
+        <div className="header">
+            <nav className="nav-bar">
+                <div className="group1">
+                    <Link to="./Home" data-testid="home"
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/Navbar.tsx]: I'd probably give this the testid "home-link-navbar" or something, as we already have a "home" test id merged in

File: WhaleSpotting/react-app/src/components/Navbar.tsx [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+                        className={currentPage === "Home" ? "nav-link-selected" : "navbar-link"}
+                        onClick={() => HandleLinkClick("Home")}>Home</Link>
+                    <Link to="./Map"
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/Navbar.tsx]:
```suggestion
                    <Link to="/Map"
```
All the links don't need the `.` AFAICT

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+    color: white;
+    text-decoration: none;  
+    font-size: calc(10px + 1vmin);  
+}
+
+.navbar-link:hover {
+    background-color: $primaryDark;
+    border-radius: 5px;
+    color: white;
+    font-size: calc(10px + 1vmin);
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: You don't need this, its already defined in the .navbar-link

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.nav-link-selected {
+    margin-left: 5%;
+    margin-right: 5%;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: Again can merge these :)

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+    padding: 3%;
+    white-space: nowrap;
+    color: white;
+    text-decoration: none;  
+    background-color: $primaryDark;
+    border-radius: 5px;  
+    font-size: calc(10px + 1vmin);
+}
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: This can be simplified to the below if you make the selected navbar items have class `"navbar-link selected"`
```suggestion
.navbar-link {
    margin: 0 5%;
    padding: 3%;
    white-space: nowrap;
    color: $textOnPrimary;
    text-decoration: none;
    font-size: $subtitleFontSize;

    &.selected, &:hover {
        color: $textOnPrimary;
        background-color: $primaryDark;
        border-radius: 5px;
    }
}
```

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+
+.dontdisplay {
+    display: none;
+}
+
+.group1 {
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: As in navbar tsx, maybe change the names of these

File: WhaleSpotting/react-app/src/tests/Navbar.test.tsx [hunk 22]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import Navbar from "../components/Navbar";
+import { BrowserRouter as Router } from "react-router-dom";
+
+test("renders navbar", () => {
+    render(<Router><Navbar /></Router>);
+    const home = screen.getByTestId("home");
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/tests/Navbar.test.tsx]: Possibly call this homeLink

**Review by oskarwilliams — APPROVED** [review]

**Review by oskarwilliams — CHANGES_REQUESTED** [review]
Meant to request changes

File: WhaleSpotting/react-app/src/styles/Navbar.scss [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+.nav-bar {
+    display: flex;
+    width: 100%;
+    max-width: 1024px;
+    flex-direction: row;
+    justify-content: space-between;
+}
+
+.navbar-link {
+    margin: 0% 5%;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Navbar.scss]: You do not need it to be 0%, it can just be 0
```suggestion
    margin: 0 5%;
```

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/react-app/src/styles/Navbar.scss (hunk 9 above)
Comment (stewartTS) [reply WhaleSpotting/react-app/src/styles/Navbar.scss]: Just to maintain a constant size across site.

Comment (oskarwilliams) [reply WhaleSpotting/react-app/src/styles/Navbar.scss]: z-index doesn't do that, it changes whether it should be on top or bottom of a view, we won't need it yet

File: WhaleSpotting/react-app/src/index.scss [hunk 24]
```diff
@@ -1,13 +1,9 @@
+@import '../src/styles/constants.scss';
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/index.scss]: this will need to have a capital C

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

**Review by oskarwilliams — APPROVED** [review]
